e2e/testcases/workload_identity_test.go

```diff
@@ -261,6 +270,16 @@ func TestWorkloadIdentity(t *testing.T) {
 				tc.sourceChart = chart.Name
 				tc.sourceVersion = chart.Version
 				tc.rootCommitFn = nomostest.HelmChartVersionShaFn(chart.Version)
+			} else if tc.sourceType == v1beta1.OciSource && tc.requireOCIArtifactRegistry {
```

Comment:
I don't think the `requireOCIArtifactRegistry` check is needed here, this variable should only really need to be used by the test scaffolding to skip the test

Reply:
The intent is to reuse the flag. Agreed it seems to be hacky.
Alternatively, we'll need another flag for building and pushing the OCI image. The sourceType isn't enough because the OCI image can be hosted in GCR.

Reply:
If that's the case, we might consider adding a `GCR` provider type to `registryproviders`. That's a bigger change though so I think this is fine for now, albeit a bit hacky. Can you leave a comment with some context?

Reply:
Sure. We can remove the check along with GCR tests once GCR is end of life.

Reply:
comment added